### PR TITLE
Allow keyfile to be passed as bytes

### PR DESCRIPTION
**Pull request by janbrummer**
In order to prevent application to create temporary files, provide keyfile as bytes.

Fixes: https://github.com/libkeepass/pykeepass/issues/363

### CODE REVIEW

**Comment by janbrummer** [comment]
@Evidlo May i ask you for a review?

File: pykeepass/kdbx_parsing/common.py [hunk 1]
```diff
@@ -116,43 +135,33 @@ def compute_key_composite(password=None, keyfile=None):
...
-                elif version.startswith('2.0'):
-                    # read keyfile data and convert to bytes
-                    keyfile_composite = bytes.fromhex(data_element.text.strip())
-                    # validate bytes against hash
-                    hash = bytes.fromhex(data_element.attrib['Hash'])
-                    hash_computed = hashlib.sha256(keyfile_composite).digest()[:4]
-                    assert hash == hash_computed, "Keyfile has invalid hash"
-        # otherwise, try to read plain keyfile
-        except (etree.XMLSyntaxError, UnicodeDecodeError):
+        if isinstance(keyfile, bytes):
```
Comment (A6GibKm) [inline pykeepass/kdbx_parsing/common.py:138]: See the save method, where it checks whether the file has `hasattr(filename, "write")` maybe the same could be done with the read attribute. BytesIO has this btw.

Comment (janbrummer) [reply pykeepass/kdbx_parsing/common.py:138]: In this case i'd need to change it from bytes to BytesIO. Not sure we want to enforce this in apps using pykeepass? NO strong opinion here, up to the reviewers i guess.

**Comment by A6GibKm** [comment]
Hello sorry to ping you @Evidlo. Is anything missing on your end?

**Comment by Evidlo** [comment]
Sorry for the late response.

I did a refactor which is a bit simpler than before, but it assumes that `keyfile` is a file-like object (such as BytesIO).

**Pull request merged**